core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2078,6 +2078,12 @@ public void delete(AlluxioURI path, DeleteContext context)
         }
 
         deleteInternal(rpcContext, inodePath, context, false);
+        if (context.getOptions().getAlluxioOnly()
+            && context.getOptions().hasDirectChildrenUnloaded()
+            && context.getOptions().getDirectChildrenUnloaded()) {
+          mInodeTree.setDirectChildrenLoaded(
+              rpcContext, inodePath.getParentInodeDirectory(), false);
+        }
```

Comment:
I feel this will be more expressive for the readers
```suggestion
        if (context.getOptions().getAlluxioOnly()
            && context.getOptions().hasSkipSyncNextTime()
            && context.getOptions().getSkipSyncNextTime()) {
          mInodeTree.setDirectChildrenLoaded(
              rpcContext, inodePath.getParentInodeDirectory(), false);
        }
```